apps/src/code-studio/components/aiTutor/aiTutorChatMessagesTable.tsx

```diff
@@ -27,8 +63,122 @@ const AITutorChatMessagesTable: React.FunctionComponent<
     })();
   }, [sectionId]);
 
+  useEffect(() => {
+    const uniqueStudentFilterOptions = chatMessages.reduce<CheckboxOption[]>(
+      (acc, message) => {
+        const userId = `${message.userId}`;
+        if (!acc.some(student => student.value === userId)) {
+          acc.push({label: message.studentName, value: userId});
+        }
+        return acc;
+      },
+      []
+    );
+    setStudentFilterOptions(uniqueStudentFilterOptions);
+  }, [chatMessages]);
+
+  const handleStatusFilterChange = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    const {value, checked} = event.target;
+    if (checked) {
+      setSelectedStatuses([...selectedStatuses, value]);
+    } else {
+      setSelectedStatuses(selectedStatuses.filter(status => status !== value));
+    }
+  };
+
+  const handleStudentFilterChange = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    const {value, checked} = event.target;
+    if (checked) {
+      setSelectedUserIds([...selectedUserIds, value]);
+    } else {
+      setSelectedUserIds(selectedUserIds.filter(student => student !== value));
+    }
+  };
+
+  const filteredChatMessages = chatMessages.filter(message => {
+    const statusMatch =
+      selectedStatuses.length === 0 ||
+      selectedStatuses.includes(message.status);
+
+    const userMatch =
+      selectedUserIds.length === 0 ||
+      selectedUserIds.includes(`${message.userId}`);
+
+    const messageDate = moment(message.createdAt);
+    let timeMatch = false;
+    switch (selectedTimeFilter) {
+      case 'lastHour':
+        timeMatch = messageDate.isAfter(moment().subtract(1, 'hours'));
+        break;
+      case 'last24Hours':
+        timeMatch = messageDate.isAfter(moment().subtract(24, 'hours'));
+        break;
+      case 'last7Days':
+        timeMatch = messageDate.isAfter(moment().subtract(7, 'days'));
+        break;
+      case 'last30Days':
+        // For now, the last 30 days and all-time filters are equivalent because of a
+        // cron job that deletes chat messages older than 30 days.
+        timeMatch = messageDate.isAfter(moment().subtract(30, 'days'));
+        break;
+      default:
+        timeMatch = true;
+    }
+
+    return statusMatch && userMatch && timeMatch;
+  });
+
+  const statusOptions = Object.values(AITutorInteractionStatus).map(status => ({
+    label: STATUS_LABELS[status] || 'Unknown',
+    value: status,
+  }));
+
   return (
     <div>
+      <div>
+        <CheckboxDropdown
+          allOptions={statusOptions}
+          checkedOptions={selectedStatuses}
+          color="black"
+          labelText="Filter by Status"
```

Comment:
Is it worth turning these into translated strings now or are things still too in flux for that? Should a follow-up task be followed for translation?

Reply:
Good call, @molly-moen. Given the quick timeline for the closed beta and uncertainty about whether we will be offering AI Tutor in languages other than English, I'd like permission to handle this as a follow-up task: https://codedotorg.atlassian.net/browse/CT-469

I've added it to the closed beta release because I think the easiest time to add the translated strings is when you write the code, but noted it as non-blocking.

Reply:
yep a follow up seems reasonable!